tools/encryption/encryptionutils/encryption-helper.go

```diff
@@ -0,0 +1,243 @@
+func encryptAllSections(parsed_html *html.Node, encrypted_sections []*html.Node, kh *keyset.Handle) error {
+	cipher, err := aead.New(kh)
+	if err != nil {
+		return err
+	}
+	for _, node := range encrypted_sections {
+		var content []string
+		for c := node.FirstChild; c != nil; c = c.NextSibling {
+			content = append(content, renderNode(c, true))
+			node.RemoveChild(c)
+		}
+		encrypted_content, encrypt_err := cipher.Encrypt([]byte(strings.Join(content, "")), nil)
+		if encrypt_err != nil {
+			return encrypt_err
+		}
+		text_node := &html.Node{Type: html.TextNode, Data: base64.StdEncoding.EncodeToString(encrypted_content)}
+	    attrs := []html.Attribute{
+			html.Attribute{Key: "type", Val: "application/octet-stream"},
+			html.Attribute{Key: "ciphertext", Val: ""},
+		}
+		script_node := &html.Node{
+			Type: html.ElementNode,
+			Data:"script",
+			DataAtom: atom.Script, 
+			Attr: attrs,
+		}
+		node.AppendChild(script_node)
+		script_node.AppendChild(text_node)
+	}
+	return nil
+}
+
+// Retrieves Google's public key from the given URL.
+func getGooglePublicKey(public_key_url string) (tinkpb.Keyset, error) {
+	resp, err := http.Get(public_key_url)
+	if err != nil {
+		return tinkpb.Keyset{}, err
+	}
+	r := keyset.NewJSONReader(resp.Body)
+	ks, err := r.Read()
+	if err != nil {
+		return tinkpb.Keyset{}, err
+	}
+	return *ks, nil
+}
+
+// Encrypts the document's symmetric key using the input Keyset.
+func encryptDocumentKey(doc_keyset string, access_requirement string, ks tinkpb.Keyset) (string, error) {
+	handle, err := keyset.NewHandleWithNoSecrets(&ks)
+	if err != nil {
+		return "", err
+	}
+	he, err := hybrid.NewHybridEncrypt(handle)
+    if err != nil {
+        return "", err
+	}
+	json_str := fmt.Sprintf("{\"accessRequirements\": [\"%s\"], \"key\": \"%s\"}", access_requirement, doc_keyset)
+	enc, err := he.Encrypt([]byte(json_str), nil)
+	if err != nil {
+		return "", err
+	}
+	return base64.StdEncoding.EncodeToString(enc), nil
+}
+
+// Adds the encrypted document key to the output document's head.
+func addEncryptedDocumentKeyToHead(encrypted_key string, parsed_html *html.Node) error {
+	for n := parsed_html.FirstChild; n != nil; n = n.NextSibling {
+		if (n.Data == "html") && (len(n.Attr) != 0) {
+			for cn := n.FirstChild; cn != nil; cn = cn.NextSibling {
+				if (cn.Data == "head") {
+					attrs := []html.Attribute{
+						html.Attribute{Key: "type", Val: "application/json"},
+						html.Attribute{Key: "cryptokeys", Val: ""},
+					}
+					crypto_keys := &html.Node{
+						Type: html.ElementNode,
+						Data:"script",
+						DataAtom: atom.Script, 
+						Attr: attrs,
+					}
+					jsonData := fmt.Sprintf(`{"google.com":"%s"}`, encrypted_key)
+					text_node := &html.Node{Type: html.TextNode, Data: jsonData}
+					crypto_keys.AppendChild(text_node)
+					cn.AppendChild(crypto_keys)
+			    	return nil
+				}
+			}
+		}
+	}
+	return fmt.Errorf("Could not add cryptokeys to head.")
```

Comment:
This also prints. Prefer to have callers handle the error. So instead return errors.New().

Reply:
Done.